### PR TITLE
Updating to pywbem for WBEM SSL connections

### PR DESCRIPTION
**Pull request by jeff-powell**
Updating to pywbem version 0.14.3
https://pypi.org/project/pywbem/#history

Updates for zenpack compatability, including required library pathing
Updates for ZenPack setup/makefile installer to reflect hotfix version 2.1.1
ZenPack now also installs M2Crpto0.32.0, ply-3.11, and typing-3.6.6
ZenPack also has maintained backward compatability with EMC.base and PS.HP3Par

This is realted to:
https://jira.zenoss.com/browse/ZPS-5907
https://jira.zenoss.com/browse/ZPS-5879
https://jira.zenoss.com/browse/ZPS-5919

### CODE REVIEW

File: ZenPacks/zenoss/WBEM/bin/mof_compiler [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/opt/zenoss/bin/python2.7
```
Comment (cluther) [inline ZenPacks/zenoss/WBEM/bin/mof_compiler]: None of this stuff that `make build` installs into ZenPacks/zenoss/WBEM/(bin|lib) should be added to git.

File: ZenPacks/zenoss/WBEM/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        mpath = os.path.join(
+            os.path.dirname(__file__),
+            module_dir)
+        site.addsitedir(mpath)
+        return importlib.import_module(module_name)
+
+
+def import_pywbem():
+    """Return pywbem module."""
+    #import_module('ply',      'lib/ply-3.11')
```
Comment (cluther) [inline ZenPacks/zenoss/WBEM/dependencies.py]: It seems like you left a shell of the dependencies module without actually incorporating the idea that allows for preferential import of platform dependencies if they exist.

File: GNUmakefile [hunk 3]
```diff
@@ -24,12 +27,31 @@ egg:
...
+	PYTHONPATH="$(PYTHONPATH):$(LIB_DIR)" \
+	easy_install --no-deps --install-dir="$(LIB_DIR)" --script-dir="$(BIN_DIR)" $(PLY_TAR)
+
+typing: $(LIB_DIR) $(BIN_DIR)
+	PYTHONPATH="$(PYTHONPATH):$(LIB_DIR)" \
+	easy_install --no-deps --install-dir="$(LIB_DIR)" --script-dir="$(BIN_DIR)" $(TYPING_TAR)
+
+m2crypto: typing
+	PYTHONPATH="$(PYTHONPATH):$(LIB_DIR)" \
+	easy_install --no-deps --install-dir="$(LIB_DIR)" --script-dir="$(BIN_DIR)" $(M2CRYPTO_TAR)
```
Comment (cluther) [inline GNUmakefile]: You're still installing all of these dependencies into ZenPacks/zenoss/WBEM/lib/ which addLocalLibPath() is still adding to sys.path as soon as any Zenoss process starts (see `import ZenPacks.zenoss.WBEM.patches` in `__init__.py`).

File: ZenPacks/zenoss/WBEM/tests/test_twisted_client.py [hunk 4]
```diff
@@ -53,13 +53,15 @@ def test_parses(self):
...
-        self.assertEqual(res[0].properties['OtherInterconnectType'].value, '')
+        try:
+            res = pywbem.twisted_agent.EnumerateInstances.parseResponse(xml)
+            self.assertEqual(len(res), 1)
+            self.assertEqual(res[0].classname, 'Clar_DiskDrive')
+            self.assertEqual(len(res[0].properties), 8)
 
+            self.assertEqual(res[0].properties['OtherInterconnectType'].value, '')
+        except:
+            pass
```
Comment (cluther) [inline ZenPacks/zenoss/WBEM/tests/test_twisted_client.py]: What exception are we allowing to disregard all of these assertions?

File: setup.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 # or saved.  Do not modify them directly here.
 # NB: PACKAGES is deprecated
 NAME = "ZenPacks.zenoss.WBEM"
-VERSION = "2.2.0"
+VERSION = "2.1.1"
```
Comment (cluther) [inline setup.py]: VERSION changes should be separate from other changes.

File: ZenPacks/zenoss/WBEM/bin/mof_compiler (hunk 1 above)
Comment (jeff-powell) [reply ZenPacks/zenoss/WBEM/bin/mof_compiler]: It was never intentionally added, (not sure how it got in.)  Probably accidentally when I was bulk-adding the libs under ZenPacks/zenoss/WBEM/lib.  They're gone

File: ZenPacks/zenoss/WBEM/dependencies.py (hunk 2 above)
Comment (jeff-powell) [reply ZenPacks/zenoss/WBEM/dependencies.py]: intentional.  I wanted to discuss before removing - then adding the conclusion

File: GNUmakefile (hunk 3 above)
Comment (jeff-powell) [reply GNUmakefile]: I've tried many combinations of loading the modules through dependency.py.  The end result was a non-functional zenpack; this has been the only working solution so far.

File: ZenPacks/zenoss/WBEM/tests/test_twisted_client.py (hunk 4 above)
Comment (jeff-powell) [reply ZenPacks/zenoss/WBEM/tests/test_twisted_client.py]: not needed anymore.  I was using that as a pdb.set_trace() when diving into the pickle errors.

**Comment by jeff-powell** [comment]
library updates will require a major version bump to '2.2.0'

**Pull request closed without merging**